src/plugins/list/list.c

```diff
@@ -0,0 +1,365 @@
+/**
+ * \file
+ *
+ * \brief Source for list plugin
+ *
+ * \copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+ *
+ */
+
+
+#ifndef HAVE_KDBCONFIG
+# include "kdbconfig.h"
+#endif
+
+#include <string.h>
+#include <kdberrors.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <kdbmodule.h>
+#include <kdbinternal.h>
+#include "list.h"
+
+typedef enum{preGetStorage = 0, postGetStorage, getEnd}GetPlacements;
+
+typedef enum{preSetStorage = 0, preCommit,	postCommit, setEnd}SetPlacements;
+
+typedef enum{preRollback = 0, postRollback, errEnd}ErrPlacements;
+
+typedef enum{GET, SET, ERR}OP;
+
+typedef struct{
+	uint8_t initialized;
+	ErrPlacements errCurrent;
+	SetPlacements setCurrent;
+	GetPlacements getCurrent;
+
+	ErrPlacements errPlacements[2];
```

Comment:
I used KeySets instead of linked lists just because they were already there and do exactly what i need. 
each KeySet is the list of plugins to be loaded for the placement.  
2 because there is just the prerollback and postrollback placement for errror.